### ISSUE TITLE
Out of Home WiFi connectivity 

### ISSUE THREAD

**Issue by killam42** [unknownskl/xbox-xcloud-client#342]
Curious if support to the stream-to home Xbox from WiFi outside the home is a possibility.

**Comment by MisterAnderson91** [comment]
Works fine from an external network from my mac to my xbox series X.

**Comment by killam42** [comment]
Oh ok then, I'm just getting a black screen on my mac when connected to anything that is my internal network.

**Comment by killam42** [comment]
Adding to this it still having a black screen when I use my phone as a hotspot

**Comment by MisterAnderson91** [comment]
Are you using v1.2.0 beta 6?

**Comment by killam42** [comment]
Yes I am, even tried to run the code direct from macOS terminal with the clone tool
Had a lot more issues because I don’t know that level of coding, but the issue remained the same

**Comment by unknownskl** [comment]
Out of home streaming is possible if the xbox is exposed to the internet. On ipv4 you will need to forward all the ports to the xbox which might be impossible to do. On IPv6 you can allow the connection to the xbox in your router / firewall and then it is possible to play over the internet using IPv6.

The official app uses Teredo to encapsulate the traffic and send it directly over the ipv4 network.

So bottom line; if you are fully IPv6 native and have no blocking firewall rules for your xbox, then yes it may probably work.
If you are bound to IPv4 then it is mostly not going to work outside the home network.

**Comment by killam42** [comment]
Thanks now I know what to start troubleshooting, but besides this issue, I'm having,
Thanks for making the app since Microsoft is being a suede bully on M1 macs by blocking the Xbox app.

**Comment by MisterAnderson91** [comment]
I can confirm that I am an running native ipv6 on my home network (which my xbox is attached to) and I am able to connect remotely without manually forwarding any ports (UPnP enabled).

**Comment by Vman7765** [comment]
Were you able to get it working killam42? I'm having the same issue as your previous post.

**Comment by killam42** [comment]
Sadly no, mainly because I could even get ipv6 working on my modem. And didn’t work even after forwarding every single Xbox port on my modem.

**Comment by MisterAnderson91** [comment]
Unless it causes the same symptoms as in this topic, you should really post it as a new issue.
